src/tree_data.c

```diff
@@ -3267,3 +3269,82 @@ lyd_get_value(const struct lyd_node *node)
 
     return NULL;
 }
+
+LIBYANG_API_DEF LY_ERR
+lyd_link_leafref_node(struct lyd_node_term *node, struct lyd_node_term *leafref_node)
```

Comment:
What is the use of all these new API functions, why not perform all those tasks automatically?

Reply:
There are  the following use-cases:
1. When model is using leafref with `required-instance false;`. In that case the validation of leafref is not happening and as such not even adding links. Therefore it has to be somehow triggered manually. I would say that calling validation is kind of non-intuitive for user and is much more complex procedure to achieve needed steps.
2. When user don't want to use the validation of libyang at all (e.g. using LYD_PARSE_ONLY), but still wish to execute checks of leafrefs and store the links.

In reality I think that we can make `lyd_link_leafref_node` and `lyd_unlink_leafref_node` as internal and keeping just `lyd_link_leafref_node_all` as public API for this use-case. What do you think about that?

Reply:
Rewritten, to have just `lyd_link_leafref_node_tree` as public API